docs/development_environment_setup.md

```diff
@@ -0,0 +1,267 @@
+# Tahoe development environment documentation
+
+The Tahoe development environment documentation setup consist in two big pieces. The edX platform Vagrant Box setup, which uses our own Appsembler branch with some customizations and the Management Console (A.K.A. AMC) setup which is a Django project running locally, but it need to connect to the edX platform box via HTTP.
+
+## edX platform vagrant box setup
+
+### Software Prerequisites
+
+Devstack requires the following software.
+
+* [VirtualBox](https://www.virtualbox.org/wiki/Downloads) 4.3.12 or later.
+* [Vagrant](https://www.vagrantup.com/downloads.html) 1.6.5 or later.
+* A Network File System (NFS) client, if your operating system does not include one. Devstack uses VirtualBox Guest Editions to share folders through NFS.
+
+### Installing Devstack with a Direct Vagrant Box Download
+1. Ensure the `nfsd` client is running.
+2. Create the `devstack` directory and navigate to it in the command prompt
+```
+mkdir devstack
+cd devstack
+```
+3. Download the Devstack Vagrant file.
+```
+curl -L https://raw.githubusercontent.com/appsembler/configuration/appsembler/ficus/master/vagrant/release/devstack/Vagrantfile.amc > Vagrantfile
+```
+
+https://github.com/appsembler/configuration/blob/appsembler/ficus/master/vagrant/release/devstack/Vagrantfile.amc
+4. Install the Vagrant vbguest plugin.
+```
+vagrant plugin install vagrant-vbguest
+```
+5. Create the Devstack virtual machine.
+```
+vagrant up
+```
+The first time you create the Devstack virtual machine, Vagrant downloads the base box, which has a file size of about 4GB. If you destroy and recreate the virtual machine, Vagrant re-uses the box it downloaded. See Vagrant’s documentation on boxes for more information.
+6. When prompted, enter the administrator password for your local computer.
+
+### Run LMS and Studio
+In order to start the LMS and Studio development servers you need to run the following commands.
+
+### Setup you hosts file to use the devstack
+Edit your `/etc/hosts` file and add the following entries:
+```
+127.0.0.1	edx.test
+127.0.0.1	studio.edx.test
+```
+
+####Connect to you vagrant box
+1. To connect to the Devstack virtual machine, use the SSH command from the devstack directory.
+```
+vagrant ssh
+```
+2. To use Devstack and perform any of the tasks described in this section, you must connect as the user edxapp and load the python env.
+```
+sudo su edxapp -s /bin/bash
+source /edx/app/edxapp/edxapp_env
+```
+
+#### LMS
+```
+paver devstack lms --settings devstack_appsembler
+```
+Open the LMS in your browser at http://edx.test:8000/
+#### Studio
+```
+paver devstack studio --settings devstack_appsembler
+```
+Open Studio in your browser at http://studio.edx.test:8001/
+
+## AMC Setup
```

Comment:
Does AMC need to run native on the OS X dev machine?

Reply:
@johnbaldwin AFAIK, you could run AMC in another VM if you wanted. It's a pretty basic django app though, so I think the idea is that it's easier to just run it native since there aren't really any tricky platform-level dependencies.

I haven't used the OS  X Postgres.app, but I imagine that would also work fine. AMC doesn't use any 3rd party database extensions or anything, so I think as long as you configure it right it should work.